enterprise/admin/__init__.py

```diff
@@ -831,22 +833,59 @@ def get_fields(self, request, obj=None):
         return fields
 
 
+class BigTableMysqlPaginator(Paginator):
+    """
+    A paginator that uses INFORMATION_SCHEMA.TABLES to estimate
+    the total number of rows in a table.
+    """
+    ARBITRARILY_LARGE_NUMBER = 10000000
+
+    # pylint: disable=attribute-defined-outside-init
+    @property
+    def count(self):  # pylint: disable=invalid-overridden-method
+        """
+        Returns the number of items in the object list (possibly an estimate).
+        """
+        query = self.object_list.query
+
+        if query.where:
+            return super().count
+        if not getattr(self, '_whole_table_count', None):
+            try:
+                with connection.cursor() as cursor:
+                    cursor.execute(
+                        'SELECT TABLE_ROWS FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = %s',
```

Comment:
Does the MySQL user for enterprise-catalog have permissions for `INFORMATION_SCHEMA` or only `enterprise_catalog` in stage/prod? I imagine it does but figured it's worth asking as I'm not too certain whether our databases have individual MySQL users per service or if they share a root MySQL user, for example.

Reply:
There are multiple MySQL/Aurora instances we use - there's a shared one that several IDAs, including enterprise-catalog are on; LMS/Studio have their own instance.  It tends to be a single user per service for all of the connections.